1.8/development/plugins/creating-modifying-templates.md

```diff
@@ -94,3 +94,46 @@ global $user;
 
 $user['favorite_colour'] = 'Blue';
 ```
+
+## Caching Templates
+Don't forget to cache the template. This is important for pages with big load and prevent the database server from extra queries.
+
+### Caching a template
+A simple way to cache an template only at the place where it's needed is to use the `THIS_SCRIPT` definition.
+For example, the template is needed in the postbit. Then we can load it in the showthreads.php.
+```php
+if(THIS_SCRIPT == 'showthread.php')
+{
+    global $templatelist;
+    if(isset($templatelist))
+    {
+        $templatelist .= ',';
+    }
+    $templatelist .= 'hello_world_template';
+}
+```
+Another example, here are some more templates needed on the member page. Then we can load it in the member.php.
+```php
+if(THIS_SCRIPT == 'member.php')
+{
+    global $templatelist;
+    if(isset($templatelist))
+    {
+        $templatelist .= ',';
+    }
+    $templatelist .= ',hello_world_template_second';
+    $templatelist .= ',hello_world_template_third';
+}
+```
+
+### Testing if the template gets cached
+You can simply test if the templates get cached correctly or not. Open the page where they should get cached and add a `?debug&debug=1` or `&debug&debug=1` to the url into the browser.
```

Comment:
You can test whether if templates get cached correctly or not with ease by opening the page where they should be cached and appending the `?debug=1` querystring to the URL. Use `&debug=1` if there are already other query strings in place. The resulting `MyBB Debug Information` page contains a table called `Page Generation Statistics` which has a row with the template cache information.

Note that only administrators can access the debug page, so ensure you are logged in with an administrator account.